### ISSUE TITLE
FFWEB-997 : Export of custom multi-attributes is not supported

### ISSUE THREAD

**Issue by michi-p** [FACT-Finder-Web-Components/magento2-module#56]
Currently, custom attributes can be added to the export feed via the Magento backend. They are added to the *Attributes* filed in the file (which contains all chosen custom attributes separated by `|`).  
e. g. 
```
" | Attribute1=Value1 | Attribute2=Value2 | Attribute3=Value3 | ... | "
```
However, if the custom attribute itself contains a multi-attribute value, it can't be imported by FACT Finder.
e. g.
```
" | Attribute1=Value1 | Attribute2= |Attribute2_1=Value2_1|Attribute2_2=Value2_2| | Attribute3=Value3 | ... | "
```
Attribute2 will be ignored at the import.

I would suggest exporting all custom attributes in separate columns in the csv.

**Comment by a-laurowski** [comment]
Hi @michi-p 
Can You give me more information what You meant by multi-attribute?
The module supports multiselect attributes but I think You have something else in mind.

**Comment by fritzmg** [comment]
And within the `Attributes` field, each attribute is also just separated by a pipe, which means each "Subattribute" cannot contain further multi-attributes.

The solution would be to change the magento2-module in such a way, that the custom attributes are simply exported as individual fields, and not within the `Attributes` field (as it used to be with the old Magento 1.9 extension from Flagbit for example).

Currently the main attribute fields are hardcoded like this:
```

    /**
     * Build a row for the product feed
     *
     * @param \Magento\Catalog\Model\Product $product
     * @param \Magento\Store\Api\Data\StoreInterface $store
     *
     * @return array
     */
    private function buildFeedRow($product, $store)
    {
        $row = [];
        $attributes = [
            'ProductNumber',
            'MasterProductNumber',
            'Name',
            'Description',
            'Short',
            'ProductUrl',
            'ImageUrl',
            'Price',
            'Manufacturer',
            'Attributes',
            'CategoryPath',
            'Availability',
            'EAN',
            'MagentoEntityId'
        ];
        foreach ($attributes as $attribute) {
            $row[$attribute] = $this->helperProduct->get($attribute, $product, $store);
        }

        return $row;
    }
```
The code should be changed, so that `$attributes` are merged with the custom defined attributes.

**Comment by fritzmg** [comment]
Also
```php
    /**
     * Get the attribute value from magento product in corresponding store
     *
     * @param $attribute
     * @param \Magento\Catalog\Model\Product $product
     * @param \Magento\Store\Api\Data\StoreInterface $store
     * @return mixed|null
     */
    public function get($attribute, $product, $store)
    {
        //max field number is 128. No Check implemented because number is fixed in this case.
        switch ((string)$attribute) {
            case "ProductNumber":
            case "MasterProductNumber":
            case "Name":
            case "Description":
            case "Short":
            case "ProductUrl":
            case "Price":
            case "CategoryPath":
            case "Availability":
            case "MagentoEntityId":
                $method = 'get' . $attribute;
                return call_user_func(array($this, $method), $product);

            case "ImageUrl":
            case "Manufacturer":
            case "Attributes":
            case "EAN":
                $method = 'get' . $attribute;
                return call_user_func(array($this, $method), $product, $store);

            default:
                return null;
        }
    }
```
of `Omikron\Factfinder\Helper\Product` would need to be changed as well, so that
```php
                $method = 'get' . $attribute;
                return call_user_func(array($this, $method), $product, $store);
```
is the default for any attribute, not just the "Attributes" or EAN.

**Comment by fritzmg** [comment]
Alternatively, the `buildFeedRow` method could fire an event with which you can add additional attributes to the `$row` yourself.

**Comment by fritzmg** [comment]
Furthermore, the current implementation exports the attributes with their translated field names. E.g.
```
Produktkennzeichnung Bio=Yes
```
instead of
```
product_label_bio=Yes
```

**Comment by fritzmg** [comment]
PR: #84

**Comment by a-laurowski** [comment]
Functionality was released in [v0.9-beta.9](https://github.com/FACT-Finder-Web-Components/magento2-module/releases/tag/v0.9-beta.9)
I'm closing this ticket
